dataprofiler/tests/profilers/test_utils.py

```diff
@@ -124,3 +124,19 @@ def test_find_diff(self):
             "g": [None, 15]
         }
         self.assertDictEqual(expected_diff, utils.find_diff_of_dicts(dict1, dict2))
+
+    def test_diff_of_dicts_with_diff_keys(self):
+        dict1 = {"unique1": 1, "shared1": 2, "shared2": 3}
```

Comment:
probably should validate the other conditions that exist in the function being tested